lib/web/git_servers.go

```diff
@@ -0,0 +1,102 @@
+/*
+ * Teleport
+ * Copyright (C) 2023  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package web
+
+import (
+	"net/http"
+
+	"github.com/gravitational/trace"
+	"github.com/julienschmidt/httprouter"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/httplib"
+	"github.com/gravitational/teleport/lib/reversetunnelclient"
+	"github.com/gravitational/teleport/lib/web/ui"
+)
+
+func (h *Handler) gitServerCreateOrUpsert(_ http.ResponseWriter, r *http.Request, _ httprouter.Params, sctx *SessionContext, site reversetunnelclient.RemoteSite) (interface{}, error) {
+	var req *ui.CreateGitServerRequest
+	if err := httplib.ReadResourceJSON(r, &req); err != nil {
+		return nil, trace.Wrap(err)
+	}
```

Comment:
Should we do a bit more validation on these web API inputs?

`ReadResourceJSON` will make sure the request body is < 1MB, but this doesn't prevent somebody from entering a really long org name or integration name, which can break gRPC listing later downstream.

Does GitHub enforce a maximum limit on org names that we can check here?

Reply:
The maximum org name is 39 chars, I can add it to:
https://github.com/gravitational/teleport/blob/108d736eec63a782585a0329bb33d589208fc245/api/types/integration_github.go#L30-L32

As for integration name, do we have any limit in general for resource name?

Reply:
Sounds good, let's validate the org name in the web API.

For integration name, I don't think we have a general limit, but we can just pick a large number - how about 2048 characters?